### PR TITLE
Store picker: Add edit button to show/hide sites

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of: #14658

## Description

This PR adds a new feature flag for hiding sites in the store picker. A new button "Edit" has also been added to the top left of the screen when the feature flag is enabled.

## Steps to reproduce

- Enable the feature flag `hideSitesInStorePicker` and build the app.
- Log in to a store and select the Menu tab.
- Open the store picker and confirm that the Edit button is available on the top right. The action for the button will be handled in a future PR.
- Disable the feature flag `hideSitesInStorePicker` and rebuild the app.
- Select the store picker again and ensure that the Edit button is no longer available.

## Testing information

Tested on simulator iPhone 16 Pro iOS 18.1 and confirm that the new Edit button is available only when the feature flag `hideSitesInStorePicker` is enabled.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/233">21.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/Authentication/StorePickerViewModelTests.swift [hunk 1]
```diff
@@ -177,6 +177,124 @@ final class StorePickerViewModelTests: XCTestCase {
         XCTAssertEqual(properties["num_of_stores"] as? Int, 2)
         XCTAssertEqual(properties["num_of_non_woo_sites"] as? Int, 1)
     }
+
+    @MainActor
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/Authentication/StorePickerViewModelTests.swift:181]: Nice tests. 👍

**Review by selanthiraiyan — APPROVED** [review]
The code LGTM and tests as expected. 🚀

**Pull request merged**